src/shared/telemetry/util.ts

```diff
@@ -85,3 +89,32 @@ export async function getUserAgent(
 
     return pairs.join(' ')
 }
+
+/**
+ * Validates that emitted telemetry metrics
+ * 1. contain a result property and
+ * 2. contain a reason propery if result = 'Failed'.
+ */
+export function validateMetricEvent(event: MetricDatum) {
```

Comment:
let's add a `fatal` parameter, then if `!fatal` we only log a warning instead of throwing an error.